### ISSUE TITLE
A `debug_type` intrinsic function

### ISSUE THREAD

**Issue by KotlinIsland** [KotlinIsland/basedmypy#248]
```py
a = 1
debug_type(a)
```
```
> mypy test.py
test.py:1:1: note: Debug type of "a":
test.py:1:1: note: upper bound: int (inferred)
test.py:1:1: note: narrowed type: Literal[1], narrowed on line 1
test.py:1:1: note: mro: int -> object
Success no errors found in 1 file
```
```py
a = 1
b = "b"
debug_type(a)
```
